### ISSUE TITLE
Helm charts deployment should merge a new release into an existing index.yaml

### ISSUE THREAD

**Issue by ringods** [samsung-cnct/k2-charts#118]
# TL;DR

Expected behaviour:
* New versions of charts must be *added* to `index.yaml` file. The `index.yaml` file should contain metadata to all versions of the charts.

Actual behaviour:
* Only 1 version of each chart exist in the `index.yaml` file. The old archives are still there, but not reachable by helm as the metadata is missing from `index.yaml`.

# Description
Yesterday, I tried deploying the `central-logging` chart on our k8s cluster, but if failed. The published `central-logging` [chart archive](http://atlas.cnct.io/central-logging-0.2.0.tgz) does not contain the archives of it's dependencies.

I suspect the release process does not contain a `helm dependency update` command when the chart version is packaged.

To circumvent that, I cloned this Git repo and ran `helm dependency update` myself. If failed because the metadata for `daemon-fluentd 0.2.0` is missing from the `index.yaml` file.

I had to bump the version in the `requirements.yaml` for the dependency update to succeed.

**Comment by ringods** [comment]
Here is how I merge a new release of a chart into an existing yaml:

```console
$ helm dependency update
$ helm package . -d ./charts --save=false
$ wget http://<repo url here>/index.yaml -O old-index.yaml
$ helm repo index ./charts --url http://<repo url here> --merge old-index.yaml
$ <publish command here>
```

**Comment by coffeepac** [comment]
@ringods thanks for the issue and for the fix.  we are in the process of overhauling our charts CI pipeline but if this is as easy at looks we may backport this one.  if not, it'll definitely go into our future state.

either way, its on the list for us now.  thanks!

**Comment by coffeepac** [comment]
we're deprecating all of k2-charts and using a completely different style of chart releases.  we will be using app registry and publishing all versions of a chart of quay.io.  

thanks for the issue and if there are any charts in this repo you use let us know!  we will be migrating a bunch to new repos but we don't support all of these so we won't be porting them to a new framework.

**Comment by ringods** [comment]
@coffeepac this app registry?

https://github.com/app-registry/appr-helm-plugin
